src/event_handler.rs

```diff
@@ -315,153 +350,159 @@ pub async fn handle_event_logs(
                 }
                 app_state.last_block_seen.store(current_block.as_u64(), Ordering::SeqCst);
 
-                // Capture the Job created event emitted by the jobs contract
-                if event.topics[0]
-                    == keccak256("JobCreated(uint256,uint8,address,bytes32,bytes,uint256,address[])")
-                    .into()
-                {
-                    // Extract the 'indexed' parameter of the event
-                    let job_id = event.topics[1].into_uint();
-                    let env = event.topics[2].into_uint();
+                // Extract the 'indexed' parameter of the event
+                let job_id = event.topics[1].into_uint();
 
-                    if env != EXECUTION_ENV_ID.into() {
-                        continue;
-                    }
+                // Extract the 'indexed' env ID and check if it's the same as executor's
+                let env_id = event.topics[2].into_uint();
+                if env_id != EXECUTION_ENV_ID.into() {
```

Comment:
Yes It is. Actually we can remove this part but then the unit test case for different environment ID will fail because the filtering happens at the event stream level itself but in test case we can only pass it directly in the stream.

Reply:
Ohh, okay. So, it's like what filters we are using isn't really getting covered in the UTs.